staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -637,6 +638,46 @@ func SearchMountPoints(hostSource, mountInfoPath string) ([]string, error) {
 	return refs, nil
 }
 
+// IsMountPoint determines if a directory is a mountpoint.
+// It should return ErrNotExist when the directory does not exist.
+// IsMountPoint is more expensive than IsLikelyNotMountPoint.
+// IsMountPoint detects bind mounts in linux.
+// IsMountPoint enumerates all the mountpoints using List() and
+// the list of mountpoints may be large, then it uses
+// isMountPointMatch to evaluate whether the directory is a mountpoint.
+func (mounter *Mounter) IsMountPoint(file string) (bool, error) {
+	isMnt, sure, isMntErr := mountinfo.MountedFast(file)
+	if sure == true && isMntErr == nil {
+		return isMnt, nil
+	}
+	if isMntErr != nil && os.IsPermission(isMntErr) {
+		// We were not allowed to do the simple stat() check, e.g. on NFS with
```

Comment:
if it is permission error, the following check should still work?

Reply:
Yes. it will work: see: https://pkg.go.dev/os#IsPermission. This should not be the case though because of the following commit in moby: https://github.com/moby/sys/commit/925ece265955e861c1292597fe0da9dd1ed66668. Note that os.Stat returns Path Error, but os.IsPermission is capably of handling one level of error.

I improved this case now so that we also account for error wrapping as well for os.Permission so that all error is properly unwrapped.

Reply:
how about the following to save a few lines?

```
if isMntErr !=nil && !errors.Is(isMntErr, fs.ErrPermission) {
    return false, err
}

```

Reply:
also as @liggitt, we should unwrap error before returning it here?

Reply:
I just addressed the two feedback pieces.

Reply:
Found one more place and addressed it as well: https://github.com/kubernetes/kubernetes/pull/109217/commits/fe5f3ef8dffc47647b510e861df03cbb6a1c4290

Here is an example: https://github.com/kubernetes/kubernetes/pull/109217/commits/fe5f3ef8dffc47647b510e861df03cbb6a1c4290

Reply:
How about using error.As() as suggested by @liggitt?

Reply:
I believe @liggitt was suggesting to unwrap to PathError but we want to return ErrNotExist which is not typed. Ref: https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/internal/oserror/errors.go;l=15-16;drc=55590f3a2b89f001bcadf0df6eb2dde62618302b 

Therefore I used this way to ensure that in case of ErrNotExist, we just return the bare error.

Reply:
As an example, please check this playground: https://go.dev/play/p/NpKmQnbhMLi where unix errors cannot be converted to PathErrors.